baldrick/plugins/github_pull_requests.py

```diff
@@ -92,38 +95,49 @@ def is_previous_comment(message):
     else:
         comment_id = comment_ids[-1]
 
-    comments = []
-    set_status = False  # Do not send a status unless we need to.
-    status = True  # True is passing
+    results = {}
     for function in PULL_REQUEST_CHECKS:
-        f_comments, f_status = function(pr_handler, repo_handler)
-        if f_status is not None:
-            set_status = True
-            status = status and f_status
-        comments += f_comments
-
-    if comments:
-        message = current_app.pull_request_prolog.format(pr_handler=pr_handler, repo_handler=repo_handler)
-        message += ''.join(comments) + current_app.pull_request_epilog
-        comment_url = pr_handler.submit_comment(message, comment_id=comment_id,
-                                                return_url=True)
-    else:
-        all_passed_message = pr_config.get("all_passed_message", '')
-        all_passed_message = all_passed_message.format(pr_handler=pr_handler, repo_handler=repo_handler)
-        if all_passed_message:
-            pr_handler.submit_comment(all_passed_message, comment_id=comment_id)
+        result = function(pr_handler, repo_handler)
+        results.update(result)
 
-        comment_url = None
-        message = ''
+    failures = [message for message, status in results.values() if not status]
+
+    if post_comment:
+
+        # Post all failures in a comment, and have a single status check
+
+        if failures:
+
+            pull_request_prologue = get_config_with_app_defaults(pr_handler, 'pull_request_prologue', default='')
```

Comment:
These config options should be under the pull request checker? i.e

```
  [ tool.baldrick.pull_requests ]
    pull_request_prologue = ''
```
?

Reply:
Also should these be `fail_prolog`?

Reply:
I don't think the comment is shown if there are no failures, so there is no such thing as a 'success' prologue?

Reply:
Yeah sure, but I just meant from a perspective of someone being able to understand the config file a bit better, if it says "failure" in the name it's obvious they are only used on fails.

Reply:
👍